RepoDb.PostgreSql/RepoDb.PostgreSql/StatementBuilders/PostgreSqlStatementBuilder.cs

```diff
@@ -399,11 +397,9 @@ public PostgreSqlStatementBuilder()
             }
 
             // Set the return value
-            var result = identityField != null ?
-                string.IsNullOrEmpty(databaseType) ?
-                    identityField.Name.AsQuoted(DbSetting) :
-                        string.Concat($"CAST({identityField.Name.AsQuoted(DbSetting)} AS {databaseType})") :
-                            primaryField != null ? primaryField.Name.AsParameter(DbSetting) : "NULL";
+            var result = identityField == null ? primaryField.Name.AsParameter(DbSetting) :
+                string.IsNullOrEmpty(databaseType) ? identityField.Name.AsQuoted(DbSetting) :
+                string.Concat($"CAST({identityField.Name.AsQuoted(DbSetting)} AS {databaseType})");
```

Comment:
Why removed the `"NULL"`? That is needed if both the `identifyField` and `primaryField` are `null`.

Reply:
first we have this 

var result = identityField != null ?
    string.IsNullOrEmpty(databaseType) ?
        identityField.Name.AsQuoted(DbSetting) :
            string.Concat($"CAST({identityField.Name.AsQuoted(DbSetting)} AS {databaseType})") :
                primaryField != null ? primaryField.Name.AsParameter(DbSetting) : "NULL";

but above we have


           if (primaryField == null)
            {
                throw new PrimaryFieldNotFoundException($"PostgreSql is using the primary key as qualifier for (INSERT or REPLACE) operation.");
            }

so `primaryField != null` is redundant.

So first refactor to if else since they are easier to deal with 

            string result;
            if (identityField != null)
                if (string.IsNullOrEmpty(databaseType))
                    result = identityField.Name.AsQuoted(DbSetting);
                else
                    result = string.Concat($"CAST({identityField.Name.AsQuoted(DbSetting)} AS {databaseType})");
            else if (primaryField != null)
                result = primaryField.Name.AsParameter(DbSetting);
            else
                result = "NULL";

again  (primaryField != null) is redundant so

       string result;
            if (identityField != null)
                if (string.IsNullOrEmpty(databaseType))
                    result = identityField.Name.AsQuoted(DbSetting);
                else
                    result = string.Concat($"CAST({identityField.Name.AsQuoted(DbSetting)} AS {databaseType})");
            else 
                result = primaryField.Name.AsParameter(DbSetting);

and != is harder to reason about for mere mortals like me

            string result;
            if (identityField == null)
                result = primaryField.Name.AsParameter(DbSetting);
            else if (string.IsNullOrEmpty(databaseType))
                result = identityField.Name.AsQuoted(DbSetting);
            else
                result = string.Concat($"CAST({identityField.Name.AsQuoted(DbSetting)} AS {databaseType})");

and your preference seems to be for collapsing ifs to ternaries and coalescing

          string result;
            if (identityField == null)
                result = primaryField.Name.AsParameter(DbSetting);
            else
                result = string.IsNullOrEmpty(databaseType) ? identityField.Name.AsQuoted(DbSetting) : string.Concat($"CAST({identityField.Name.AsQuoted(DbSetting)} AS {databaseType})");

and again

     var result = identityField == null ? primaryField.Name.AsParameter(DbSetting) :
                string.IsNullOrEmpty(databaseType) ? identityField.Name.AsQuoted(DbSetting) :
                string.Concat($"CAST({identityField.Name.AsQuoted(DbSetting)} AS {databaseType})");